sdk/communication/azure-communication-identity/src/test/java/com/azure/communication/identity/CommunicationIdentityAsyncTests.java

```diff
@@ -92,6 +97,132 @@ public void createUserAndToken() {
             .verifyComplete();
     }
 
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("com.azure.communication.identity.TokenCustomExpirationTimeHelper#getValidExpirationTimes")
+    public void createUserAndTokenWithValidCustomExpiration(String testName, Duration tokenExpiresIn) {
+        // Arrange
+        CommunicationIdentityClientBuilder builder = createClientBuilder(httpClient);
+        asyncClient = setupAsyncClient(builder, "createUserAndTokenWithValidCustomExpiration" + testName);
+        List<CommunicationTokenScope> scopes = Arrays.asList(CommunicationTokenScope.CHAT);
+
+        // Action & Assert
+        Mono<CommunicationUserIdentifierAndToken> createUserAndToken = asyncClient.createUserAndToken(scopes, tokenExpiresIn);
+        StepVerifier.create(createUserAndToken)
+            .assertNext(result -> {
+                assertNotNull(result.getUserToken());
+                assertNotNull(result.getUser());
+
+                if (getTestMode() == TestMode.LIVE) {
+                    CommunicationIdentityClientUtils.TokenExpirationDeviationData tokenExpirationData =
+                        CommunicationIdentityClientUtils.tokenExpirationWithinAllowedDeviation(tokenExpiresIn, result.getUserToken().getExpiresAt());
+                    assertTrue(tokenExpirationData.getIsWithinAllowedDeviation(),
+                        CommunicationIdentityClientUtils.getTokenExpirationOutsideAllowedDeviationErrorMessage(tokenExpiresIn,
+                            tokenExpirationData.getActualExpirationInSeconds())
+                    );
+                }
+            })
+            .verifyComplete();
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("com.azure.communication.identity.TokenCustomExpirationTimeHelper#getInvalidExpirationTimes")
+    public void createUserAndTokenWithInvalidCustomExpiration(String testName, Duration tokenExpiresIn) {
+        // Arrange
+        CommunicationIdentityClientBuilder builder = createClientBuilder(httpClient);
+        asyncClient = setupAsyncClient(builder, "createUserAndTokenWithInvalidCustomExpiration" + testName);
```

Comment:
```suggestion
        asyncClient = setupAsyncClient(builder, "createUserAndTokenWithInvalidCustomExpiration " + testName);
```
for test logs readability :)